mypy/test/testmerge.py

```diff
@@ -64,6 +65,7 @@ def run_case(self, testcase: DataDrivenTestCase) -> None:
 
         main_src = '\n'.join(testcase.input)
         messages, manager, graph = self.build(main_src)
+        assert manager is not None, 'cases where CompileError occurred should not be run'
```

Comment:
This is a disturbing situation. Perhaps we should (in a later PR) fix the API for build.build() so it always returns and translates CompileError into a regular result (assuming we can't completely get rid of CompileError, which is debatable)?